aeneas/src/vst/Verifier.v3

```diff
@@ -1687,6 +1726,11 @@ class TypeChecker(ERROR: ErrorGen, file: VstFile) extends VstVisitor<Type, Type>
 			else UnresolvedOp(expr.op.token, ltype, null);
 			return getErrorType();
 		}
+		if (op.opcode == Opcode.RefEq && !TypeSystem.isSubtype(ltype, rtype) &&
```

Comment:
I don't think you need this extra case; it should handle incomparable types by default already.

Reply:
If so, a `TypeError` instead of a `Comparison Error `is generated in test case `cmp02.v3`. Should I change the test case?

Reply:
Yeah, please update the test case.

Reply:
test case updated